terracumber-cli

```diff
@@ -314,6 +345,10 @@ def main():
         print("Running terraform...")
         results['terraform'] = run_terraform(args, tf_vars)
 
+    if args.sumaform_backend == 'aws' and (args.runstep == 'cucumber' or args.custom_repositories):
```

Comment:
Let's not make the bastion related to the backend being used. It should work as well if we add Azure tomorrow. And it should work if for whatever reason we don't use a bastion.

I'd suggest you check if `get_bastion_hostname` is returning `None` when the bastion does not exist, and if so you check here if it's needed or not.

You will also need to run when `args.runall` is true.